js-user-registration-service/src/test/java/com/jumia/warmup/jsuserregistrationservice/rest/UserRegistrationRestTest.java

```diff
@@ -0,0 +1,259 @@
+package com.jumia.warmup.jsuserregistrationservice.rest;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.jumia.warmup.jsuserregistrationservice.dtos.AccountInformationDTO;
+import com.jumia.warmup.jsuserregistrationservice.dtos.ContactsDTO;
+import com.jumia.warmup.jsuserregistrationservice.dtos.PersonalDetailsDTO;
+import com.jumia.warmup.jsuserregistrationservice.dtos.UserDTO;
+import com.jumia.warmup.jsuserregistrationservice.services.UserRegistrationService;
+import com.jumia.warmup.jsuserregistrationservice.utils.Constants;
+import com.jumia.warmup.jsuserregistrationservice.utils.Constants.ErrorMessages;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(UserRegistrationRest.class)
+public class UserRegistrationRestTest {
```

Comment:
As this test is for the `UserRegistrationRestController` and uses the mockMVC to perform a POST API call this is a sort of integration test (although it's mainly aimed at asserting different validation rules for User) 

We should have a separate Unit Test for each DTO created to assert data is set correctly and follows the validation rules at a lower level than this.

Reply:
Done.